index.ts

```diff
@@ -1,35 +1,23 @@
+// index.ts
 import { ApolloServer } from 'apollo-server';
+import { resolvers } from './resolvers/user-resolver';
+import { typeDefs } from './types/types';
+import prisma from './prisma';
 
-// Pegar uma porta personaliada do ENV
-const PORT = process.env.PORT || 4000;
-
-// Definindo o Schema do GraphQL
-const typeDefs = `
-type User {
-    firstname: String
-    lastname: String
-    age: Int
-    address: String
-}
-type Query {
-    users: [User]
-    hello: String
-}
-`;
-
-const resolvers = {
-  Query: {
-    users: () => [],
-    hello: () => `Hello, world`,
-  },
-};
-
-// Instancia do ApolloServer
 const server = new ApolloServer({
   typeDefs,
   resolvers,
 });
 
-server.listen(PORT).then(({ url }) => {
-  console.log(`Server ready at ${url}`);
-});
+(async () => {
```

Comment:
legal que isolou o setup do server aqui, julgo que seria um bom primeiro passo tirar o `const server = ...` desse arquivo, seguindo na linha do comentário do Alan, logo logo vc vai precisar exportar esse `server` e é um problema ele estar aqui, porque ai na hora que for importar o `server` vai rodar de novo esse trecho aqui.

Via de regra deixaria esse código apenas inicializando o servidor e qualquer declaração pode ficar em outro arquivo isolado (só com a declaração ou inicialização)